src/nodes.coffee

```diff
@@ -2871,19 +2870,17 @@ exports.StringWithInterpolations = class StringWithInterpolations extends Base
               match
         fragments.push @makeCode value
       else
-        fragments.push @makeCode '$' if not @csx
+        fragments.push @makeCode '$' unless @csx
         code = element.compileToFragments(o, LEVEL_PAREN)
-        code = @wrapInBraces code if not @isNestedTag element
+        code = @wrapInBraces code unless @isNestedTag element
         fragments.push code...
-    fragments.push @makeCode '`' if not @csx
+    fragments.push @makeCode '`' unless @csx
     fragments
 
   isNestedTag: (element) ->
-    @csx and
-    (exprs = element?.body?.expressions) and
-    exprs.length is 1 and
-    (call = exprs?[0]) instanceof Call and
-    call.csx
+    exprs = element?.body?.expressions
+    call = exprs?[0]
+    @csx and exprs and exprs.length is 1 and call instanceof Call and call.csx
```

Comment:
@GeoffreyBooth I'd prefer to revert this. It has different semantics, and although it should be ok, the point was that this is only happening for `@csx`. If you want to keep the style, add an early return.